File: app/src/main/java/org/sitcon/ccip/network/webclient/WebChromeViewClient.java
package org.sitcon.ccip.network.webclient;

import android.view.View;
import android.webkit.WebChromeClient;
import android.webkit.WebView;
import android.widget.ProgressBar;

public class WebChromeViewClient extends WebChromeClient {
    ProgressBar progressBar;

    public WebChromeViewClient(ProgressBar progressBar) {
        this.progressBar = progressBar;
    }

    @Override
    public void onProgressChanged(WebView view, int newProgress) {
        super.onProgressChanged(view, newProgress);
        setWebProgress(newProgress);
    }

    public void setWebProgress(int progress) {
        progressBar.setVisibility(View.VISIBLE);
        progressBar.setProgress(progress);
        if (progress == 100) {
            progressBar.setVisibility(View.GONE);
        }
    }
}

File: app/src/main/java/app/opass/ccip/network/webclient/WebChromeViewClient.kt
package app.opass.ccip.network.webclient

import android.view.View
import android.webkit.PermissionRequest
import android.webkit.WebChromeClient
import android.webkit.WebView
import android.widget.ProgressBar

class WebChromeViewClient(private var progressBar: ProgressBar, private var permissionRequestCallback: (request: PermissionRequest?)->Unit) : WebChromeClient() {
    override fun onProgressChanged(view: WebView, newProgress: Int) {
        super.onProgressChanged(view, newProgress)
        setWebProgress(newProgress)
    }

    override fun onPermissionRequest(request: PermissionRequest?) {
        permissionRequestCallback(request)
    }

    private fun setWebProgress(progress: Int) {
        progressBar.visibility = View.VISIBLE
        progressBar.progress = progress
        if (progress == 100) {
            progressBar.visibility = View.GONE
        }
    }
}
